src/rooted_tree.cpp

```diff
@@ -9,6 +9,47 @@ RootedTree::RootedTree(const Tree& tree) : Tree(tree.Topology(), tree.BranchLeng
          "root. Are you trying to parse unrooted trees into a RootedSBNInstance?");
 }
 
+void RootedTree::InitializeParameters(
+    const std::unordered_map<Tag, double>& taxon_date_map) {
+  size_t leaf_count = LeafCount();
+  parameters_ = std::vector<double>(leaf_count - 1, -1);
+
+  node_heights_ = std::vector<double>(Topology()->Id() + 1);
+  node_bounds_ = std::vector<double>(Topology()->Id() + 1);
+
+  std::vector<double> branch_lengths = BranchLengths();
+  int root_id = static_cast<int>(Topology()->Id());
+
+  // Set up bounds and heights
+  // Start with the leaves from the dates map
+  for (const auto& pair : taxon_date_map) {
```

Comment:
Remember that in C++17 we can introduce meaningful names and avoid first and second with things like
```
for (const auto &[begin, end] : subsplit_ranges) {
```